components/datetime/src/provider/date_time.rs

```diff
@@ -281,7 +282,12 @@ impl DateTimeSymbols for provider::calendar::DateSymbolsV1 {
                         _ => widths.abbreviated.as_ref(),
                     };
                     if let Some(symbols) = symbols {
-                        return &symbols.0[(day as usize) % 7];
+                        let idx = (day as usize) % 7;
+                        return symbols
+                            .0
+                            .get(idx)
+                            .map(|x| &**x)
```

Comment:
`optional (n-b)`: can you use `.map(|x| x.as_ref())` here? It's more readable.

Reply:
It won't work since we're dereferencing a Cow here, deref coercions don't work through iterators like that

Reply:
Did you check? I pulled your branch, replaced with `x.as_ref()` and it compiles.

Reply:
`.map(AsRef::as_ref)` works as well